### ISSUE TITLE
Linter: Implement `a11y-no-redundant-image-alt` rule

### ISSUE THREAD

**Issue by marcoroth** [marcoroth/herb#1224]
Implement the [`GitHub::Accessibility::NoRedundantImageAlt`](https://github.com/github/erblint-github/blob/main/lib/erblint-github/linters/github/accessibility/no_redundant_image_alt.rb) rule from [`erblint-github`](https://github.com/github/erblint-github).

### Rule Details

`<img>` alt prop should not contain "image" or "picture" as screen readers already announce the element as an image.

This rule does not discourage conveying the _medium_ of the image (e.g. "screenshot", "painting", "photograph") which may be important to help a user better understand the content.

### Examples

**Incorrect** code for this rule:

```html
<img alt="picture of Mona Lisa" src="monalisa.png">
<img alt="image of a fluffy dog" src="dog.png">
```

**Correct** code for this rule:

```html
<img alt="Mona Lisa" src="monalisa.png">
<img alt="The original painting of Mona Lisa" src="monalisa.png">
<img alt="Screenshot of the Settings page" src="settings_page.png">
```

### References

- [erblint-github source](https://github.com/github/erblint-github/blob/main/lib/erblint-github/linters/github/accessibility/no_redundant_image_alt.rb)
- [erblint-github docs](https://github.com/github/erblint-github/blob/main/docs/rules/accessibility/no-redundant-image-alt.md)
- [W3C WAI Images Tutorial](https://www.w3.org/WAI/tutorials/images/)
- [Primer: Alternative text for images](https://primer.style/design/accessibility/alternative-text-for-images)

### Parent Issue

- #1205

**Comment by brunoprietog** [comment]
This is true, but Do you think it makes sense to implement it in this context? For example, this would fail anyway if you use dynamic text, text in other languages, or something like the `t` helper. Although, I don't think it actually gives false positives either.

**Comment by marcoroth** [comment]
Yeah this is a good point. I think for this rule it makes sense to assume that we are dealing with static content in the HTML attribute and we'd consider any ERB expression as "accessible" since we can't statically analyze it. 

Though the plan is that, eventually, we can also run the Herb Linter via ReActionView on your rendered HTML view and report these offenses that way. Since we know where each HTML element is coming from we can precisely map it back to the source template!

**Comment by brunoprietog** [comment]
That would be awesome!
